api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentService.java

```diff
@@ -53,17 +60,17 @@ public Payment makePayment(PaymentCycle paymentCycle, String cardAccountId) {
 
     private Payment createPayment(PaymentCycle paymentCycle, String cardAccountId, Integer amountToPay) {
         return Payment.builder()
-                    .cardAccountId(cardAccountId)
-                    .claim(paymentCycle.getClaim())
-                    .paymentAmountInPence(amountToPay)
-                    .paymentCycle(paymentCycle)
-                    .build();
+                .cardAccountId(cardAccountId)
+                .claim(paymentCycle.getClaim())
+                .paymentAmountInPence(amountToPay)
+                .paymentCycle(paymentCycle)
+                .build();
     }
 
-    private Integer checkBalanceAndCalculatePaymentAmount(PaymentCycle paymentCycle) {
-        // TODO: HTBHF-1267: Check balance against card provider, update paymentCycle with balance and timestamp
-        // TODO: HTBHF-1267: reduce amount paid if it would put the card over the max allowed balance
-        return paymentCycle.getTotalEntitlementAmountInPence();
+    private Integer checkBalanceAndCalculatePaymentAmount(PaymentCycle paymentCycle, String cardAccountId) {
+        CardBalanceResponse balance = cardClient.getBalance(cardAccountId);
+        paymentCycleService.updateAndSavePaymentCycleWithBalance(paymentCycle, balance);
+        return paymentCalculator.calculatePaymentAmountCycleInPence(paymentCycle.getVoucherEntitlement(), balance.getAvailableBalanceInPence());
```

Comment:
I think we possibly made a mistake in the payment calculator. Should method name be
```
paymentCalculator.calculatePaymentCycleAmountInPence()
```
?

Reply:
agreed :)

Reply:
Updated